feat: implementing xy parser (#4)

kjappelbaum opened the pull request:
should address #4

Commit:
feat: implementing xy parser (#4)

chore: run CI from any branch

fix: CI from any branch

fix: forgot to push test file

lpatiny commented on src/__tests__/parser.test.js:
```diff
@@ -28,3 +28,20 @@ describe('parseDiffractogram', () => {
+  it('check the dictionary', () => {
+    const diffractogram = parseXY(xy);
+    expect(diffractogram.meta.userName).toBe('Lab Manager');
+    expect(diffractogram.data.x[0]).toStrictEqual(10);
```
Checking length of x and y array would make sense

lpatiny commented on src/parser/parser.js:
```diff
@@ -92,3 +93,47 @@ function getXYDiffractogram(data) {
+  let tmp = s.split(':');
+  return tmp[1].replace('"', '').replace("'", '').trim();
+}
+export function parseXY(file, options = {}) {
```
Better to have various files in the 'parser' folder
* parseXY.js
* parseBRML.js ?? (not sure about the name because it is a part of BRML right ?

In the same folder add a '__tests__' with the corresponding test
* parseXY.test.js
* parseBRML.test.js

lpatiny commented on src/parser/parser.js:
```diff
@@ -92,3 +93,47 @@ function getXYDiffractogram(data) {
+  return tmp[1].replace('"', '').replace("'", '').trim();
+}
+export function parseXY(file, options = {}) {
+  let lines = file.split('\n');
```
If prefer to filter the result for empty lines. You never know how many there are.
```js
let lines=file.split('\n').filter( line => ! line.match(/^\s*$/))
```

lpatiny commented on src/parser/parser.js:
```diff
@@ -92,3 +93,47 @@ function getXYDiffractogram(data) {
+  let lines = file.split('\n');
+  const header = lines[0];
+  lines.splice(0, 1); // header line
+  lines.splice(-1, 1); // last empty line
```
Remove this line if empty line are filtered before

lpatiny commented on src/__tests__/parser.test.js:
```diff
@@ -28,3 +28,20 @@ describe('parseDiffractogram', () => {
   });
 });
+
+describe('parseXY', () => {
```
this will move in src/parser/__tests__/parseXY.test.js

lpatiny commented on src/index.js:
```diff
@@ -1,17 +1,29 @@
-  const jcamp = fromJSON(result.data, { meta: result.meta, info: result.info });
-
-  return jcamp;
+async function xrdConverter(binary, filetype = 'brml', options = {}) {
```
Should it be 'fileType' ? camelCase ?

I would put it as a property of options and then

const {fileType='brml'} = options;

kjappelbaum replied:
thanks! You see i come from Python and just recently started JS  :D

lpatiny commented on src/index.js:
```diff
@@ -1,17 +1,29 @@
-
-  return jcamp;
+async function xrdConverter(binary, filetype = 'brml', options = {}) {
+  if (filetype.match('brml')) {
```
Please already use a 
```
switch (filetype.toLowerCase()) {
  case 'brml':

...
```

Commit:
chore: refactored.

moving brml and xy parsing routines in seperate files

Commit:
chore: linting.

lpatiny reviewed (changes requested):
I think that we could get rid of the 'reader' folder and only have 'parser'. Like that we have

parseBRML
parseXY

Now if you agree you could move the code in 'xrd-analysis'. In all those kind of projects I try to have a 'from' and a 'to' folder. If there are more feature like peak picking or analysis of the spectra we could then do something like

const spectrum = XRDAnalysis.fromXY(textFile);
XRDAnalysis.peakPicking(spectrum);

To convert to jcamp we have something like
const analysis = XRDAnalysis.fromXY(textFile);
const jcamp =toJcamp(analysis);


We could have a generic 'from' depending the filename but I don't think this should be manage at the level of the library.

Commit:
chore: first refactoring step. Getting grid of the "reader" folder

chore: linting.

chore: moved the tests.

fix: fixing the tests.

chore: do not delete last line, instead strip all of them.

chore: switch to switch

chore: code simplification, adressing #8

kjappelbaum commented:
Again, thanks for your comments! Most of the points should be fixed. I'll clean up for moving it to `xrd-analysis`

Commit:
chore: fix filename

Commit:
chore: updated readme with note that xrd-analysis should be used.
